yrs/src/block.rs

```diff
@@ -1387,45 +1433,88 @@ impl ItemContent {
         }
     }
 
-    /// Returns a formatted content of an item. For complex types (represented by [BranchRef] nodes)
-    /// it will return a target type of a branch node (eg. Yrs [Array], [Map] or [XmlElement]). For
-    /// other types it will returns a vector of elements stored within current block. Since block
-    /// may describe a chunk of values within it, it always returns a vector of values.
-    pub fn get_content(&self) -> Vec<Value> {
-        match self {
-            ItemContent::Any(v) => v.iter().map(|a| Value::Any(a.clone())).collect(),
-            ItemContent::Binary(v) => vec![Value::Any(Any::Buffer(v.clone().into_boxed_slice()))],
-            ItemContent::Move(_) => Vec::default(),
-            ItemContent::Deleted(_) => Vec::default(),
-            ItemContent::Doc(_, v) => vec![Value::Any(*v.clone())],
-            ItemContent::JSON(v) => v
-                .iter()
-                .map(|v| Value::Any(Any::String(v.clone().into_boxed_str())))
-                .collect(),
-            ItemContent::Embed(v) => vec![Value::Any(v.as_ref().clone())],
-            ItemContent::Format(_, _) => Vec::default(),
-            ItemContent::String(v) => v
-                .chars()
-                .map(|c| Value::Any(Any::String(c.to_string().into_boxed_str())))
-                .collect(),
-            ItemContent::Type(c) => {
-                let branch_ref = BranchPtr::from(c);
-                vec![branch_ref.into()]
+    /// Reads a contents of current [ItemContent] into a given `buf`, starting from provided
+    /// `offset`. Returns a number of elements read this way (it cannot be longer than `buf`'s len.
+    pub fn read(&self, offset: usize, buf: &mut [Value]) -> usize {
```

Comment:
A new way of reading the contents of the `ItemContent` - previously we had `get_content`, `content_first` and `content_last` to read the all block values or the first/last value. With this we can provide a buffer of fixed length and read block values at any given offset we like.